### PR TITLE
Update Prometheus vendoring and simplify remote append code

### PR DESCRIPTION
**Pull request by juliusv**

### CODE REVIEW

**Comment by juliusv** [comment]
The remote.WriteResponse doesn't exist anymore and was never really used anyways.

**Comment by juliusv** [comment]
More importantly, the vendoring update brings back the chunk `Utilization()` methods, which I'll reintroduce to Prism in the next PR.

File: server.go [hunk 1]
```diff
@@ -126,22 +126,6 @@ func AppenderHandler(appender SampleAppender) http.Handler {
...
-		}
-
-		if _, err := snappy.NewWriter(w).Write(respBuf); err != nil {
-			log.Errorf("write err: %v", err)
-			http.Error(w, err.Error(), http.StatusInternalServerError)
-			return
-		}
-
-		w.Header().Add("Content-Encoding", "snappy")
-		w.WriteHeader(http.StatusOK)
```
Comment (jml) [inline server.go]: Does this mean we're not returning a response code any more? That seems undesirable. Perhaps we should retain the status line?

Comment (juliusv) [inline server.go]: Nope, `http.StatusOK` is the default.

Will merge when it's not weekend (cause of CD).

**Pull request merged**